core/src/main/java/org/jboss/set/mjolnir/archive/mail/MailingService.java

```diff
@@ -0,0 +1,73 @@
+package org.jboss.set.mjolnir.archive.mail;
+
+import javax.annotation.Resource;
+import javax.ejb.Stateless;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.*;
+
+@Stateless
```

Comment:
IMO this is in fact a stateful bean, since we can modify it's state via the setter methods. This being `@Stateless` would cause issues if the bean were used by several clients simultaneously, e.g. if we executed several reporting jobs in parallel (although that won't probably happen) - they could be overriding this bean's data under each other's hands.

From an "architecture" PoV it might be cleaner to pass the data as arguments of the `send()` method (or a single wrapper object argument). That way we would eliminate the setters and we could keep this as a `@Stateless` or `@Standalone` EJB.

Reply:
Or even better it would be just a CDI `@ApplicationScoped` bean, rather than EJB. Unless `@Resource` injections work only in EJBs? I'm not sure.